p2p/server.go

```diff
@@ -660,6 +660,8 @@ func (srv *Server) setupDiscovery() error {
 	if err != nil {
 		return err
 	}
+	defer conn.Close()
```

Comment:
I didn't find the place where the close is called (or should be called) yet, but calling it when `setupDiscovery` returns is too early, since real usage only starts afterwards.

Reply:
that't make sense, but we should find a good place for closing it

Reply:
Maybe it is being closed properly. Is there a specific reason to believe otherwise?

Reply:
the main reason to make this change for me is because it can make another issue when nodes go shutdown
if we don't make sure all listening ports and other stuff is closed gracefully and correctly 
it can cause some issue like port already listening or make running node again hard
at all i'm not expert in blockchain and network and maybe it can be wrong idea